ui/src/app/applications/components/applications-list/applications-filter.tsx

```diff
@@ -229,4 +218,57 @@ export class ApplicationsFilter extends React.Component<ApplicationsFilterProps,
         }
         return `${cluster.name} (${cluster.server})`;
     }
+
+    private updateLabels(labels: Map<string, Set<string>>, app: models.Application) {
```

Comment:
I think I found a way to avoid duplicating filtering logic. The same behavior can be achieved if we change line 337 in application-list.tsx:
https://github.com/argoproj/argo-cd/blob/991894e30535f6289fb76efa7a4f355e04a35dfa/ui/src/app/applications/components/applications-list/applications-list.tsx#L337

to

```javascript
applications={filterApps(applications, pref, pref.search)}
```

perhaps in make sense to save result of `filterApps` into variable and reuse result to improve performance.

Reply:
Heh, it was as simple as that.  @alexmt, ok, I've made the changes.

I had to also reuse `applications`.  Ready for second review.